src/main/java/io/milvus/client/AbstractMilvusGrpcClient.java

```diff
@@ -417,704 +164,1389 @@ private FieldData genFieldData(String fieldName, DataType dataType, List<?> obje
         add(DataType.BinaryVector);
     }};
 
+    private void waitForLoadingCollection(String collectionName, List<String> partitionNames,
+                                          long waitingInterval) throws IllegalResponseException {
+        if (partitionNames == null || partitionNames.isEmpty()) {
+            ShowCollectionsRequest showCollectionRequest = ShowCollectionsRequest.newBuilder()
+                    .addCollectionNames(collectionName)
+                    .setType(ShowType.InMemory)
+                    .build();
+
+            // Use showCollection() to check loading percentages of the collection.
+            // If the inMemory percentage is 100, that means the collection has finished loading.
+            // Otherwise, this thread will sleep a small interval and check again.
+            while (true) {
```

Comment:
what about timeout? do we need a timeout for waiting to load?

Reply:
same as flush, we should have a time limit of blocking wait, for example 5 mins.

Reply:
Sounds good. I will add a parameter to control the time out.